dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -871,8 +886,30 @@ public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Parameter(hidden
     public Set<Entry> getStarredServices(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user) {
         User u = userDAO.findById(user.getId());
         checkNotNullUser(u);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Service)
-                .collect(Collectors.toCollection(LinkedHashSet::new));
+        return this.getStarredEntries(u, EntryType.SERVICE);
+    }
+
+    /**
+     * Helper function for getStarred endpoints
+     *
+     * @param user
+     * @param entryType
+     * @return a user's starredEntries for desired entryType
+     */
+    public Set<Entry> getStarredEntries(User user, EntryType entryType) {
+        if (entryType == EntryType.WORKFLOW) {
+            return user.getStarredEntries().stream().filter(BioWorkflow.class::isInstance)
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+        } else if (entryType == EntryType.SERVICE) {
+            return user.getStarredEntries().stream().filter(Service.class::isInstance)
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+        } else if (entryType == EntryType.NOTEBOOK) {
+            return user.getStarredEntries().stream().filter(Notebook.class::isInstance)
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+        } else { // else would either be Tool or AppTool
+            return user.getStarredEntries().stream().filter(element -> element instanceof Tool || element instanceof AppTool)
```

Comment:
This one is tricker, but this may help 
``` 
Predicate<Entry> predicate = BioWorkflow.class::isInstance;
Predicate<Entry> otherPredicate = element -> (element instanceof Tool || element instanceof AppTool);
```